lib/dpl/provider/pages.rb

```diff
@@ -60,17 +74,31 @@ def github_deploy
         context.shell "echo '#{@gh_fqdn}' > CNAME" if @gh_fqdn
         context.shell 'git add .'
         context.shell "git commit -m 'Deploy #{@project_name} to #{@gh_ref}:#{@target_branch}'"
-        context.shell "git push --force --quiet 'https://#{@gh_token}@#{@gh_ref}' master:#{@target_branch} > /dev/null 2>&1"
+        if @gh_token
+          target_repo = "https://#{@gh_token}@#{@gh_ref}"
+        else
+          # Depend on the key set up in push_app.
+          target_repo = "git@github.com:#{slug}.git"
+        end
+        context.shell "git push --force --quiet '#{target_repo}' master:#{@target_branch}"
```

Comment:
I don't absolutely need to remove the `> /dev/null 2>&1` for this change, but it makes errors much easier to debug.

Reply:
@jyasskin I've added `--debug` support (AFAIR natively supported by dpl) to my feature branch for this :)